src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp

```diff
@@ -2155,6 +2171,293 @@ class at_icc_start_frostwing_gauntlet : public AreaTriggerScript
         }
 };
 
+class npc_flesh_eating_insect : public CreatureScript
+{
+public:
+    npc_flesh_eating_insect() : CreatureScript("npc_flesh_eating_insect") { }
+
+    struct npc_flesh_eating_insectAI : public ScriptedAI
+    {
+        npc_flesh_eating_insectAI(Creature* creature) : ScriptedAI(creature)
+        {
+            me->SetReactState(REACT_AGGRESSIVE);
```

Comment:
I believe it's because they're critters, but now that you mention it, critters have this interesting behavior that they always start fleeing once they take damage (and I don't see it being handled here)

https://github.com/TrinityCore/TrinityCore/blob/d7c53c30fb45f6da015e126aeb8a4a1901f72e58/src/server/game/AI/CoreAI/PassiveAI.cpp